x/dsmr/node_test.go

```diff
@@ -994,11 +1057,220 @@ func TestNode_NewBlock_IncludesChunkCerts(t *testing.T) {
 	}
 }
 
+// TestDuplicateChunksElimination tests that duplicate chunks that have appeared before are getting correctly eliminated.
+func TestDuplicateChunksElimination(t *testing.T) {
+	r := require.New(t)
+
+	node := newTestNode(t)
+
+	blk := Block{
+		ParentID:  ids.GenerateTestID(),
+		Height:    1,
+		Timestamp: 1,
+		blkID:     ids.GenerateTestID(),
+	}
+	r.NoError(node.Accept(context.Background(), blk))
+
+	_, chunkCert, err := node.BuildChunk(
+		context.Background(),
+		[]tx{
+			{
+				ID:     ids.GenerateTestID(),
+				Expiry: 4,
+			},
+		},
+		4,
+		codec.Address{},
+	)
+	r.NoError(err)
+
+	blk = Block{
+		ParentID:  ids.GenerateTestID(),
+		Height:    2,
+		Timestamp: 2,
+		blkID:     ids.GenerateTestID(),
+		ChunkCerts: []*ChunkCertificate{
+			&chunkCert,
+		},
+	}
+	r.NoError(node.Accept(context.Background(), blk))
+
+	_, err = node.BuildBlock(context.Background(), blk, 3)
+	r.ErrorIs(err, ErrNoAvailableChunkCerts)
+
+	// make sure that it's not the case with any other chunk.
+	_, _, err = node.BuildChunk(
+		context.Background(),
+		[]tx{
+			{
+				ID:     ids.GenerateTestID(),
+				Expiry: 4,
+			},
+		},
+		4,
+		codec.Address{},
+	)
+	r.NoError(err)
+	r.NoError(node.Accept(context.Background(), blk))
```

Comment:
I feel like instead of having a single test case that checks:
1. Build a duplicate chunk, have it fail, and build a new chunk and see if it passes

It would just make more sense to have this split into separate test cases:
1. Build a Chunk, it should pass
2. Build a duplicate chunk, it should fail
3. Build a chunk, build a duplicate chunk, should pass with a single chunk in the block